openssl/src/encrypt.rs

```diff
@@ -157,6 +158,32 @@ impl<'a> Encrypter<'a> {
         }
     }
 
+    /// Sets the RSA OAEP label.
+    ///
+    /// This is only useful for RSA keys.
+    ///
+    /// This corresponds to [`EVP_PKEY_CTX_set0_rsa_oaep_label`].
+    ///
+    /// [`EVP_PKEY_CTX_set0_rsa_oaep_label`]: https://www.openssl.org/docs/manmaster/man3/EVP_PKEY_CTX_set0_rsa_oaep_label.html
+    #[cfg(any(ossl102, libressl310))]
+    pub fn set_rsa_oaep_label(&mut self, label: CString) -> Result<(), ErrorStack> {
```

Comment:
This should just take a `&str` rather than a CString.

Reply:
How about a `&[u8]`? Or a `&CStr`?

Reply:
My spec (https://trustedcomputinggroup.org/wp-content/uploads/TCG_TPM2_r1p59_Part1_Architecture_pub.pdf section 4) ask for quite specific sequence of bytes, and ideally I'd like to stick as close as possible to it.

Reply:
A string is composed of a specific sequence of bytes, but if OAEP labels are not constrained to be textual, `&[u8]` seems fine.

Reply:
Yeah, I know. I'm not too sure what's expected and what's not for an oaep label to be fair. I'll try to dig up some documentation.

Reply:
https://tools.ietf.org/html/rfc8017#section-2
defines the OAEP label as:
> L              optional RSAES-OAEP label, an octet string

https://tools.ietf.org/html/rfc8017#section-4
then defines the octet string as:
> For the purposes of this document, and consistent with ASN.1 syntax,
   an octet string is an ordered sequence of octets (eight-bit bytes).
   The sequence is indexed from first (conventionally, leftmost) to last
   (rightmost).  For purposes of conversion to and from integers, the
   first octet is considered the most significant in the following
   conversion primitives.

So my opinion is that it should be an `&[u8]`.

Reply:
Cool - that sounds right to me as well.